transcom/mymove
# B 20010 int Add FF for HQ role

## [B-20010](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A949820)

## Summary

The changes here are to FF off the Headquarters Role in the Admin App Office Account creation page and remove the HQ role checkbox from the Office App Request Account page. Since FF can't currently be read without being signed in the Office App will get the HQ role checkbox back once we determine all HQ role options are functioning properly. For now the only way to create a HQ user role is via the Admin App Office Creation page with the FEATURE_FLAG_HEADQUARTERS_ROLE feature flag turned on.

This backlog adds a new Headquarters Office user role. Further downstream tasks will add privileges meant for the HQ role which will limit pages to read only.

Is there anything you would like reviewers to give additional scrutiny?

Turn the FEATURE_FLAG_HEADQUARTERS_ROLE on/off and test. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login to the admin application
2. Create a new Office user
3. Select the Safety Moves Privilege then attempt to select HQ checkbox you should NOT be able to
4. Deselect Safety Moves Privilege then select HQ role
5. Attempt to select Safety Moves Privilege you should NOT be able to
6. Select the TOO role and click Save

![image (2)](https://github.com/transcom/mymove/assets/149000222/c9ed8189-5b41-4247-91cd-69d95c4f57d9)

7. Go to the Office app and sign in as the newly created user
8. Click the change user role and verify you can login with the HQ role
![image (3)](https://github.com/transcom/mymove/assets/149000222/a82392e1-f07e-4189-b475-469ed5b26637)
NOTE: At this point you should be able to do what a TOO can do but this story was to just allow a new login role

9. **Turn off** the **FEATURE_FLAG_HEADQUARTERS_ROLE** by changing it to false in the **.envrc** file
10. Also turn off **headquarters_role** in the **development.features.yaml** file by setting **enabled** and **value** to **false** for the headquarters_role key.
11. Kill your server and client in the terminal
12. Run direnv allow
13. Relaunch server and client in terminal
14. Login to the Admin app and go to create a new Office User
15. Fill out the form then attempt to select the Headquarters role checkbox you should NOT be able to

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
